lib/fluent/msgpack_factory.rb

```diff
@@ -23,25 +23,32 @@ module MessagePackFactory
 
     module Mixin
       def msgpack_factory
-        if $log
-          $log.warn('Deprecated method: this method is going to be deleted. Use Fluent::MessagePackFactory.engine_factory')
+        unless @deprecated_log_done
```

Comment:
This is just my curiosity. Feel free to ignore it.
Using single flag is reasonable for preventing duplicated warning logs?

Reply:
I think printing at least one warning is enough becuase the point of this warning is "Do not use Fluent::MsgpackFactory::Mixin".

Reply:
Sounds reasonable.